osh/client/commands/cmd_find_tasks.py

```diff
@@ -47,10 +49,30 @@ def options(self):
             action="store_true",
             help="query by NVR (default one)"
         )
+        self.parser.add_option(
+            "-s",
+            "--states",
+            action="append",
+            type="string",
+            nargs=1,
+            help=(f"query by task state. This option is used in conjunction with -r, -p, or -n. "
+                  f"Specify multiple states by using it multiple times, like '-s FAILED -s CLOSED'. "
+                  f"Valid choices include {', '.join(s.upper() for s in TASK_STATES)}.")
+        )
+
+    def _validate_states(self, states):
+        """
+        State validator, raise error in case user specifies invalid task states
+        """
+        valid_states = [state.lower() for state in TASK_STATES]
+        invalid_states = [s for s in states if s.lower() not in valid_states]
```

Comment:
If upper case is going to be the default, the code can be further simplified like this:
```suggestion
                  f"Valid choices include {', '.join(TASK_STATES)}.")
        )

    def _validate_states(self, states):
        """
        State validator, raise error in case user specifies invalid task states
        """
        valid_states = list(TASK_STATES)
        invalid_states = [s for s in states if s.upper() not in valid_states]
```

Reply:
updated